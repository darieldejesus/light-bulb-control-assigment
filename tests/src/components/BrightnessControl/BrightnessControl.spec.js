import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

import api from '../../../../src/helpers/ApiClient';
import clientMiddleware from '../../../../src/stateManager/middleware/client';
import ReduxBrightnessControl, { BrightnessControl } from '../../../../src/components/BrightnessControl';
import theme from '../../../../src/theme';

const middleware = [
  clientMiddleware(api),
  thunk,
];
const mockStore = configureMockStore(middleware);

describe('BrightnessControl', () => {
  const props = {
    active: true,
    updateDeviceBrightness: jest.fn(),
    id: 5,
    brightness: 10,
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BrightnessControl {...props} />);
  });

  test('Should render the brightness control', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Should call "updateDeviceBrightness" with the Id and Brightness', () => {
    const {
      active,
      updateDeviceBrightness,
      id,
      brightness,
    } = props;
    wrapper.props().onChange(brightness);
    expect(updateDeviceBrightness).toBeCalledWith(id, {
      brightness,
      active,
    });
  });

  test('Should dispatch "updateBrightness" with the Id and Brightness', () => {
    const store = mockStore({
      devices: {
        isUpdatingName: false,
      },
    });
    const customProps = {
      active: true,
      id: 5,
      brightness: 10,
    };
    wrapper = mount(
      <Provider store={store}>
        <ReduxBrightnessControl {...customProps} theme={theme} />
      </Provider>,
    );
    const {
      active,
      id,
      brightness,
    } = customProps;
    const expectedAction = [
      {
        type: 'resinio/devices/UPDATE',
      },
    ];
    wrapper.find('BrightnessControl').props().updateDeviceBrightness(id, {
      active,
      brightness,
    });
    expect(store.getActions()).toEqual(expectedAction);
    // expect(updateDeviceBrightness).toBeCalledWith(id, {
    //   brightness,
    //   active,
    // });
  });
});
