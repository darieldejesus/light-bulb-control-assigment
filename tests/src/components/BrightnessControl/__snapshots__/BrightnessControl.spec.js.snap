// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrightnessControl Should render the brightness control 1`] = `
<SliderSelectControl
  circleEndAngle={315}
  circleStartAngle={45}
  enabled={true}
  onChange={[Function]}
  padding={50}
  radius={120}
  strokeWidth={13}
  value={10}
/>
`;
